core/src/consensus/solo/mod.rs

```diff
@@ -121,6 +121,9 @@ impl ConsensusEngine for Solo<CodeChainMachine> {
             self.machine.add_balance(block, &address, reward)?;
         }
         self.machine.increase_term_id(block, last_term_finished_block_num)?;
+
+        let current_term = block.state().metadata()?.unwrap().current_term_id();
+        stake::on_term_close(block.state_mut(), current_term)?;
```

Comment:
The last block changes the term. In other words, the term changes after executing the last block of the term, not before the first block of the next term. So you should call `on_term_close` with the changed term id.
Please fix the test cases.

Reply:
What about
```rust
        let current_term = block.state().metadata()?.unwrap().current_term_id();
        self.machine.increase_term_id(block, last_term_finished_block_num)?;
        stake::on_term_close(block.state_mut(), current_term)?;
```

Reply:
Looks good, but it seems that `on_term_close` doesn't use the `last_term_finished_block_num`, so the order is not important.
I made it increase the term id after `on_term_close` and there is no need to pass the term id because `on_term_close` takes the state.